src/main/java/org/elasticsearch/river/RiversService.java

```diff
@@ -214,37 +211,6 @@ public void riverClusterChanged(RiverClusterChangedEvent event) {
                 if (routing == null || !localNode.equals(routing.node())) {
                     // not routed at all, and not allocated here, clean it (we delete the relevant ones before)
                     closeRiver(riverName);
-                    // also, double check and delete the river content if it was deleted (_meta does not exists)
-                    try {
-                        client.prepareGet(riverIndexName, riverName.name(), "_meta").setListenerThreaded(true).execute(new ActionListener<GetResponse>() {
-                            @Override
-                            public void onResponse(GetResponse getResponse) {
-                                if (!getResponse.isExists()) {
-                                    // verify the river is deleted
-                                    client.admin().indices().prepareDeleteMapping(riverIndexName).setType(riverName.name()).execute(new ActionListener<DeleteMappingResponse>() {
-                                        @Override
-                                        public void onResponse(DeleteMappingResponse deleteMappingResponse) {
-                                            // all is well...
-                                        }
-
-                                        @Override
-                                        public void onFailure(Throwable e) {
-                                            logger.debug("failed to (double) delete river [{}] content", e, riverName.name());
-                                        }
-                                    });
-                                }
-                            }
-
-                            @Override
-                            public void onFailure(Throwable e) {
-                                logger.debug("failed to (double) delete river [{}] content", e, riverName.name());
-                            }
-                        });
-                    } catch (IndexMissingException e) {
-                        // all is well, the _river index was deleted
-                    } catch (Exception e) {
-                        logger.warn("unexpected failure when trying to verify river [{}] deleted", e, riverName.name());
-                    }
```

Comment:
I was wrong. To remove a _river we should run:

       DELETE _river/rivername/_meta